test/es6/classes_bugfixes.js

```diff
@@ -326,6 +326,24 @@ var tests = [
         new class0("cat", 100, {});
     }
   },
+  {
+    name: "Issue1586: Out of stack for super.<method>.apply",
+    body: function () {
+        class A {
+            m() { return 'A'; }
+            static n() { return 'B'; }
+        }
+
+        class B extends A {
+            m() { return super.m.apply(this); }
+            static n() { return super.n.apply(this); }
+        }
+
+        var b = new B();
+        assert.areEqual('A', b.m(), "");

```

Comment:
updated.

---
In reply to: [79953187](https://github.com/Microsoft/ChakraCore/pull/1613#discussion_r79953187) [](ancestors = 79953187)